monitoring/iostats_context.cc

```diff
@@ -11,17 +11,18 @@ namespace ROCKSDB_NAMESPACE {
 
 #ifdef ROCKSDB_SUPPORT_THREAD_LOCAL
 __thread IOStatsContext iostats_context;
-#endif
+#else
+// Should not be used because the counters are not thread-safe.
+// Put here just to make get_iostats_context() simple without ifdef.
+static IOStatsContext iostats_context;
+#endif  // !ROCKSDB_SUPPORT_THREAD_LOCAL
 
 IOStatsContext* get_iostats_context() {
-#ifdef ROCKSDB_SUPPORT_THREAD_LOCAL
   return &iostats_context;
-#else
-  return nullptr;
-#endif
 }
```

Comment:
Should this return nullptr if NIOSTATS_CONTEXT is defined?  Should it assert?  Or can we make the function not exist at all of NIOSTATS_CONTEXT is defined?

Reply:
- It does not have to and should not nullptr. If user defines `NIOSTATS_CONTEXT` explicitly (off by default), it indicates the user knows what to expect. It is better than returning a nullptr without user knowing about it, as reported in https://github.com/facebook/mysql-5.6/issues/904. If we allow returning nullptr, then user has to do nullptr-check every time, just to be safe. This is more expensive, especially for profiling code.

- It should not assert either, because assertion is no-op in opt mode.

- We do not have to make the function not exist. If we should, then I will argue that the entire file be made non-existent if `NIOSTATS_CONTEXT` is defined. So does this apply to perf context. However, this is not necessary.